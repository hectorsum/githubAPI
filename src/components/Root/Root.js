import React from 'react'
import { useFetchCommits } from '../../hooks/useFetchCommits'
import { getPetTime } from '../../helpers/getPetTime';

export const Root = () => {
  const {commits:{data},loading} = useFetchCommits();
  if (loading){
    console.log('loading');
  }else{
    console.log(getPetTime(data));
  }
  
  return (
    <div>
      <h1>Github API</h1>
    </div>
  )
}
