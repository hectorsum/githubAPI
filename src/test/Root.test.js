import React from "react";
import "@testing-library/jest-dom";
import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { Root } from "../components/Root/Root";
import { createSerializer } from "enzyme-to-json";

configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Testing <Root/>", () => {
  const wrapper = shallow(<Root />);
  test("Should show <Root/> component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
