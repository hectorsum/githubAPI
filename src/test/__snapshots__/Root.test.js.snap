// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Root/> Should show <Root/> component correctly 1`] = `
<BrowserRouter>
  <AppRouter />
</BrowserRouter>
`;
